broker/src/main/java/io/camunda/zeebe/broker/system/configuration/backup/BackupStoreCfg.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.system.configuration.backup;
+
+import io.camunda.zeebe.broker.system.configuration.ConfigurationEntry;
+
+public class BackupStoreCfg implements ConfigurationEntry {
+
+  private BackupStoreType store = BackupStoreType.NONE;
+
+  private S3BackupStoreConfig s3 = new S3BackupStoreConfig();
```

Comment:
Here, I tried to reuse `io.camunda.zeebe.backup.s3.S3BackupConfig`, but it did not work. I assume it is because it is a record and the config cannot be parsed into a record. I expected it to work out of the box. Since it did not work, a new configuration class defined.

Reply:
Hm, according to the Spring documentation it shouldn't matter anymore, as long as there is only one constructor (see https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-2.6-Release-Notes#records-and-configurationproperties)

I guess because we have two constructors it doesn't work =/ Could we make the second constructor a static factory method and try again?

Reply:
It did not work after removing the second constructor in `S3BackupConfig`. So I will leave it as it is.